stages/org.osbuild.dmverity

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+import os
+import subprocess
+import sys
+import tempfile
+import osbuild.api
+
+
+def open_dmverity(blocksize, partlabel, datapart, hashpart, mountpoint):
+    hash = subprocess.run(["/usr/sbin/veritysetup",
+                            "format", datapart, hashpart,
+                            "--data-block-size", f"{blocksize}"],
+                           check=True, capture_output=True)
+    for line in hash.stdout.decode().splitlines():
+      if line.startswith("Root hash:"):
+        hash = line.split(":", 1)[1].strip()
+        break
+
+    subprocess.run(["/usr/sbin/veritysetup",
+                    "open", datapart, partlabel, hashpart, hash],
+                   check=True)
+    subprocess.run(["mount", "-o", "ro", f"/dev/mapper/{partlabel}", mountpoint],
+                   check=True)
+    return hash
+
+
+def close_dmverity(partlabel, mountpoint):
+    subprocess.run(["umount", mountpoint], check=False)
+    subprocess.run(["/usr/sbin/veritysetup", "close", partlabel], check=False)
+
+
+def main(tree, paths, devices, options):
+    blocksize = options.get("blocksize", 512)
+
+    disk = os.path.join(paths["devices"], devices["disk"]["path"])
+    partition = f"{disk}p{options["partition"]}"
```

Comment:
Maybe all this "data_partition" to be symetric with "hash_partition" (it's also closer to what the veritysetup man page calls it)?